File: liqp-core/src/main/java/liqp/filters/Capitalize.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Capitalize extends Filter {

    /*
     * (Object) capitalize(input)
     *
     * capitalize words in the input sentence
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        String original = super.asString(value);

        if (original.isEmpty()) {
            return original;
        }

        char first = original.charAt(0);

        return Character.toUpperCase(first) + original.substring(1);
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Capitalize.kt
package liqp.filter

import liqp.context.LContext
import liqp.params.FilterParams

/**
 * {{ 'bob jones' | capitalize }} -> Bob jones
 */
class Capitalize : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return context.asString(value)?.capitalize() ?: value
  }
}
